### PR TITLE
Bjd tools

### PR DESCRIPTION
**Pull request by mpyat2**
This branch is referred to #286.

What's new:

1) A new "Tool" plug-in "JD to BJD": a dialog that allows conversion of a list of JD or HJD dates to BJD_TDB ones. For this, it utilizes the https://astroutils.astronomy.osu.edu/time/ service.

2) "BJD_TDB Converter", similar to the "Heliocentric JD converter". Converts datasets with JD or HJD dates into BJD_TDB. It also uses https://astroutils.astronomy.osu.edu/time/

3) Better "Coordinates" dialog in "Heliocentric JD converter" (the same as in the new "BJD_TDB Converter"). For data loaded from a file or sources other than AAVSO ID, it asks for coordinates and has a "VSX Name" button which gets coordinates from VSX.

@dbenn , please look at them. Probably, you'll have some suggestions. I will make additional checks and let you know when I finish.

### CODE REVIEW

**Comment by dbenn** [comment]
This all sounds awesome. Started having a look. Thanks Max.

**Comment by dbenn** [comment]
@mpyat2 I think what would make sense is for all of this to be merged to master when you think you are ready. Then I'd like to revisit the idea of HJD converter being an observation transformer plug-in. The BJD converter could take that form as well but I'd like to start with the HJD converter by first adding  plugin test, then converting it and checking the results are the same. This will give us undoable edits. If that works out, you or I could do the same thing for the BJD converter.

So, in summary, I suggest we get all this great work merged, then do any further work on another branch, initially for https://github.com/AAVSO/VStar/issues/84.

What do you think?

**Comment by dbenn** [comment]
@mpyat2 One observation to make that I'm sure you are aware of is the amount of time it takes to run, because of the web request latency I imagine. I don't see how that can be improved right now though. All good though. It's much better to have this functionality than not to have it.

**Review by dbenn — APPROVED** [review]
I'm inclined to say at this point that we should merge when you are happy and then we can revisit the question of observation transformer plugin. I'll start with a HJD plugin test and converting that under https://github.com/AAVSO/VStar/issues/84 as mentioned in another comment.

**Comment by mpyat2** [comment]
Hi @dbenn 
I've done some additional checks, and I think the request is ready.

**Review by dbenn — COMMENTED** [review]
`ConfirmDialogWithHelp` and `NumberFieldHelper` at least look like they could be moved into the VStar core code. `getCoordinates()` too? Anything else?

Now or in a later refactoring?

**Comment by dbenn** [comment]
Thanks Max. Let me know if you have questions about comments. In general looks awesome.

**Comment by mpyat2** [comment]
> `ConfirmDialogWithHelp` and `NumberFieldHelper` at least look like they could be moved into the VStar core code. `getCoordinates()` too? Anything else?
> 
> Now or in a later refactoring?

Yes, surely, it's better to move it to the core code. Better do it in a later refactoring, I think. The small steps are safer :)

File: plugin/src/org/aavso/tools/vstar/external/plugin/BJDConverter.java [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+					
+				}
+			} else {
+				MessageBox.showWarningDialog("BJD_TDB Converter",
+						"Canceled by user: the previously loaded observations have NOT been converted to BJD_TDB.");
+			}
+		}
+	}
+
+	private boolean showConfirmDialog2(String title, String msg, String helpTopic) {
```
[the remark this answers is not in the document]
Comment (mpyat2) [reply plugin/src/org/aavso/tools/vstar/external/plugin/BJDConverter.java]: > Could probably change to showConfirmDialog()

Done!

**Comment by mpyat2** [comment]
Hi @dbenn,

There were conflicts in build.xml (I added BJD tools, and you changed the baseline shifter to the mag. and baseline shifter).
I tried to fix them, if something goes wrong after the commit, I'll check build.xml and fix it, I suppose it is not critical for now.
